atc/db/volume.go

```diff
@@ -462,25 +472,26 @@ func (volume *createdVolume) initializeResourceCache(tx Tx, resourceCache Resour
 		if pqErr, ok := err.(*pq.Error); ok && pqErr.Code.Name() == pqUniqueViolationErrCode {
 			// another volume was 'blessed' as the cache volume - leave this one
 			// owned by the container so it just expires when the container is GCed
-			return false, nil
+			return nil, false, nil
 		}
 
-		return false, err
+		return nil, false, err
 	}
 
 	affected, err := rows.RowsAffected()
 	if err != nil {
-		return false, err
+		return nil, false, err
 	}
 
 	if affected == 0 {
-		return false, ErrVolumeMissing
+		return nil, false, ErrVolumeMissing
 	}
 
+	volume.workerResourceCacheID = workerResourceCache.ID
```

Comment:
Before this PR, it uses `resource cache id` + `worker name` to look for `worker resource cache`, it supposed to have a single tuple found by the criteria. Now, that criteria cannot uniquely identify a worker resource cache, because it may be 1 valid cache and multiple invalidated caches. Thus I changed to use  `workerResourceCache.ID` to identify a specific cache.